src/wizard/interactive_build.jl

```diff
@@ -136,44 +136,101 @@ end
 """
 function interactive_build(state::WizardState, prefix::Prefix,
                            ur::Runner, build_path::AbstractString;
-                           hist_modify = string)
-   histfile = joinpath(prefix, "metadir", ".bash_history")
-   runshell(ur, stdin=state.ins, stdout=state.outs, stderr=state.outs)
-   # This is an extremely simplistic way to capture the history,
-   # but ok for now. Obviously doesn't include any interactive
-   # programs, etc.
-   if isfile(histfile)
-       state.history = hist_modify(state.history,
+                           hist_modify = string, srcdir_overlay = true)
+    histfile = joinpath(prefix, "metadir", ".bash_history")
+    cmd = `/bin/bash -l`
+
+    if srcdir_overlay
+        cmd = """
+            mount -t overlay overlay -o lowerdir=/workspace/srcdir_lower,upperdir=/workspace/srcdir,workdir=/workspace/srcdir_work /workspace/srcdir
+            cd \$(pwd)
+            /bin/bash --login
+            """
+        cmd = `/bin/bash -c $cmd`
+    end
+
+    run_interactive(ur, cmd, stdin=state.ins, stdout=state.outs, stderr=state.outs)
+
+    had_patches = false
+    if srcdir_overlay
+        # Check if there were any modifications made to the srcdir
+        if length(readdir(joinpath(prefix, "srcdir"))) > 0
+            # Note that we're mounting to /workspace/srcdir/upper so comparing
```

Comment:
Is this comment right?  It looks like you're mounting to `/workspace/srcdir`, not `/workspace/srcdir/upper`.